octobot_trading/personal_data/portfolios/portfolio_value_holder.py

```diff
@@ -256,7 +268,52 @@ def convert_currency_value_using_last_prices(self, quantity, current_currency, t
             pass
         except KeyError:
             pass
-        raise errors.MissingPriceDataError(f"no price data to evaluate {current_currency} price in {target_currency}")
+        raise errors.MissingPriceDataError(
+            f"no price data to evaluate {current_currency} price in {target_currency}")
+
+    def _try_converting_with_multiple_pairs(
+        self, currency, quantity) -> decimal.Decimal:
+        # settlement_asset needs to be handled to add support for futures
+                
+        # try with two pairs
+        # for example:
+        # currency: ETH - ref market: USDT
+        #                   BTC/ETH      ->    BTC/USDT
+        # first covert ETH -> BTC and then BTC -> USDT
+        for symbol_str in symbol_data.get_config_symbols(
+            self.portfolio_manager.exchange_manager.config, True
+        ):
+            parsed_symbol = symbol_util.parse_symbol(symbol_str)
+            if parsed_symbol.base == currency:
+                first_ref_market = parsed_symbol.quote
+                second_symbol_str = symbol_util.merge_currencies(
+                        first_ref_market, self.portfolio_manager.reference_market
+                        )
+                # check if second pair is available
+                # first pair is already known to be available
+                if self.portfolio_manager.exchange_manager.symbol_exists(
+                    second_symbol_str
+                    ):
+                    try:
+                        first_ref_market_value = (
+                            self.convert_currency_value_using_last_prices(
+                                quantity, currency, first_ref_market
+                            )
+                        )
+                        if first_ref_market_value:
+                            ref_market_value = (
+                                self.convert_currency_value_using_last_prices(
+                                    first_ref_market_value,
+                                    first_ref_market,
+                                    self.portfolio_manager.reference_market,
+                                )
+                            )
+                            if ref_market_value:
+                                return ref_market_value
+                    except errors.MissingPriceDataError:
+                        # conversion pairs might not ne initialized
+                        # or is not available at all
+                        continue # trying with other pairs
```

Comment:
```suggestion
                        pass # trying with other pairs
```
(continue will skip any other lines after it. Even though there is no line here, it's not necessary, pass is what we want to do I think

Reply:
done